INTERNAL.md

```diff
@@ -14,55 +14,63 @@ All terminologies are the unique names for clear understanding.
 includes function registry and type registry.
 
 ## System Context
- Context is shared across all cluster nodes. Once loaded from the system
- configuration and the system constants, Context is always kept as immutable.
+A context is shared across all cluster nodes. Once loaded from the system
+configuration and the system constants, Context is always regarded as immutable.
 
 Session is a set of metadata overriding context.
 
 ## Default Context
 
 ## Job Context
-Job Context is a final context for each job execution. It will be shared 
+Job Context is a final context for each job execution. It will be shared
 across all Tasks for a single job.
 
 # In-memory row data structure
-In this section, we address how rows and columns are represented through 
-in-memory data structure in Tokamak core. A minipage is just a sequence 
-memory block. A single row or some column values can be stored in a minipage. 
-We can implement row-oriented or column-oriented approaches with the a single 
-data structure. A page is a container for a list of minipages. So, a page 
-represents multiple rows in the row store as well as column store.
+In this section, we address how rows and columns are stored in in-memory data
+structures in Tokamak core. A minipage is a sequence of memory blocks.
+A single row or some column values can be stored in a minipage. As a result,
+either row-oriented or column-oriented approaches can be implemented with a
+single data structure. A page is a container for a list of minipages. So,
+a page represents multiple rows in row stores as well as column stores.
 
 ## MiniPage details
 
 ### Fixed-length (direct) MiniPage
-A fixed-length minipage implementation allows different encoding ways. 
-Basically, uncompressed minipage is supported, and later we will implement 
+A fixed-length minipage implementation allows different encoding ways.
+Basically, uncompressed minipage is supported, and later we will implement
 some light-weight encoding minipages, such as:
  * Run-length encoded minipage
  * Bit-vector encoded minipage
 
 ### Variable-length MiniPage
-For a variable length field or huge-sized values like BLOB, MiniPage supports 
-indirect minipage, just storing pointers in the minipage and each pointer 
-indicates an actual field value. Another application of indirect minipage 
-is dictionary encoding.
+For a variable length field or huge-sized values like BLOB, MiniPage provides
+an indirect way to access its data, i.e., just pointers to data are stored in
+minipages and each pointer indicates an actual field value. Another application
+of indirect minipage is dictionary encoding.
 
 # JIT Compilation
-(TODO)
-In Tokamak, JitManager generates some JIT code fragments in runtime for 
+In Tokamak, JitManager generates some JIT code fragments in runtime for
 arithmetic, comparison, other logical expressions and predicates.
 
 Access to nested fields
 
-Hash function
+* Various invocation methods
```

Comment:
It moved and is separated into multiple parts. See the following lines.

https://github.com/hyunsik/tokamak/pull/1/files#diff-0fa472b032003b2960b08cb7545de6b0R114

https://github.com/hyunsik/tokamak/pull/1/files#diff-0fa472b032003b2960b08cb7545de6b0R120